substate-store-impls/src/hash_tree/entity_tier.rs

```diff
@@ -33,7 +33,7 @@ impl<'s, S: ReadableTreeStore> EntityTier<'s, S> {
     }
 }
 
-impl<'s, S> Tier for EntityTier<'s, S> {
+impl<'s, S> TierView for EntityTier<'s, S> {
```

Comment:
(naming) I confirm the need to fight the "Node" name-clash... Entity would be almost perfect, but: 
- it feels it applies to higher levels of abstraction in Radix model,
- in the impl, it still suddenly turns into a "node" again (e.g. near `DbNodeKey`, no matter if you alias it),
- "Entity", as a word, is just vague in almost every context (kinda like naming something "Object"),

Since the "node" problem is broader (remember that we also have a Java app named "Node" 🤦), I suggest to fight it with prefixes:
- ReNode (a storage-level concept representing an Entity)
- NetworkNode (the app)
- TreeNode (an atom of a Tree)

Reply:
So `Entity` is our public-facing name for any persisted RENode, and is used across our APIs and Docs. As the JMT is persisted, I think it is fine to call it entity.

In fact, possibly just renaming `DbNodeKey` to `DbEntityKey` might even be correct :D.